doc/source/roadmap-detailed.rst

```diff
@@ -0,0 +1,367 @@
+    failures) - improvements in testing and build documentation at least are
+    needed.
+
+Continuous integration is in good shape, it covers Windows, macOS and Linux, as well
+as a range of versions of our dependencies and building release quality wheels.
+
+
+Modules
+-------
+
+cluster
+```````
+This module is in good shape.
+
+
+constants
+`````````
+This module is basically done, low-maintenance and without open issues.
+
+
+fftpack
+```````
+We aim to follow NumPy in adopting ``pocketfft`` (see `this NumPy PR
+<https://github.com/numpy/numpy/pull/11888>`__).  That will address a number of
+maintenance issues, and increase performance (both accuracy and speed).
+Of particular interest regarding performance is the Bluestein algorithm (or
+chirp Z-transform), which we have been wanting to add to ``fftpack`` for a long
+time.
+
+We probably want to deprecate fftpack.convolve as public function (it was not
+meant to be public).
+
+There's a large overlap with ``numpy.fft``.  This duplication has to change
+(both are too widely used to deprecate one); in the documentation we should
+make clear that ``scipy.fftpack`` is preferred over ``numpy.fft``.
+If there are differences in signature or functionality, the best version
+should be picked case by case (example: numpy's ``rfft`` is preferred, see
+gh-2487).
+
+Ideas for new features:
+
+- Add a backend/plugin system.  At the moment pyFFTW is monkeypatching SciPy,
+  and ``mkl_fft`` provides ``fftpack``-compatible functions as well.  We should
+  provide a method to support such packages.
+
+integrate
+`````````
+Needed for ODE solvers:
+
+  - Documentation is pretty bad, needs fixing
+  - A new ODE solver interface  (``solve_ivp``) was added in SciPy 1.0.0.
+    In the future we can consider (soft-)deprecating the older API.
+
+The numerical integration functions are in good shape.  Support for integrating
+complex-valued functions and integrating multiple intervals (see gh-3325) could
+be added.
+
+
+interpolate
+```````````
+
+Ideas for new features:
+
+  - Spline fitting routines with better user control.
+  - Integration and differentiation and arithmetic routines for splines
+  - Transparent tensor-product splines.
+  - NURBS support.
+  - Mesh refinement and coarsening of B-splines and corresponding tensor products.
+
+io
+``
+wavfile;
+
+    - PCM float will be supported, for anything else use ``audiolab`` or other
+      specialized libraries.
+    - Raise errors instead of warnings if data not understood.
+
+Other sub-modules (matlab, netcdf, idl, harwell-boeing, arff, matrix market)
+are in good shape.
+
+
+linalg
+``````
+``scipy.linalg`` is in good shape.  We have started requiring more recent
+LAPACK versions (minimum version increases from 3.1.0 to 3.4.0 in SciPy 1.2.0);
+we want to add support for newer features in LAPACK.
+
+Needed:
+
+  - Reduce duplication of functions with ``numpy.linalg``, make APIs consistent.
```

Comment:
where can we start this discussion? Mailing list?

Reply:
I think it's mostly about "making APIs consistent". We can't remove much, there will be duplicate functions. I'd say the important thing is to have an overview of where APIs don't match, and then figure out which one is better. One example that is discussed in an issue somewhere: the modes of `qr` don't (or didn't?) match. 

Note, I just moved this content, didn't think about it recently.